### DISCUSSION TITLE
Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use.

### DISCUSSION THREAD

**Discussion by ahmedRSA** [AdguardTeam/AdGuardHome#5652] [Q&A] [answered]
running this command giving this error

```bash
sudo docker run --name adguardhome\
    --restart unless-stopped\
    -v /my/own/workdir:/opt/adguardhome/work\
    -v /my/own/confdir:/opt/adguardhome/conf\
    -p 53:53/tcp -p 53:53/udp\
    -p 67:67/udp -p 68:68/udp\
    -p 80:80/tcp -p 443:443/tcp -p 443:443/udp -p 1337:1337/tcp\
    -p 853:853/tcp\
    -p 784:784/udp -p 853:853/udp -p 8853:8853/udp\
    -p 5443:5443/tcp -p 5443:5443/udp\
    -d adguard/adguardhome
```

### error
```
docker: Error response from daemon: driver failed programming external connectivity on endpoint adguardhome (78d64ae66417bc96176c11ca4d6d3529682b0086ffaa26c8c75b024488817389): Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use.
```
### sudo lsof -nP | grep LISTEN
![image](https://user-images.githubusercontent.com/77731877/228516150-e100c25c-1972-493e-a9a4-7f7c6a2566e9.png)

### nginx config [dns.DOMAIN.in.conf]
```js
server { listen 80; server_name dns.DOMAIN.in;

location / { 
    proxy_pass http://localhost:1337; 
}}
```

**Comment by Pamalosebi** [comment, marked as answer]
I have the same problem. However, it worked with Docker 20.10.23-143.
With 24.0.2-144 it stopped working. The error message is identical to yours.

Other containers seem to work fine (also with port configs). So I am not sure if it is Docker or AdGuardHome.

**Reply by Pamalosebi** [reply]
I have found a solution here: https://stackoverflow.com/questions/38517272/docker-proxy-error-starting-userland-proxy-while-trying-to-bind-on-443

Edit `/etc/docker/daemon.json` and add
```
{
    "userland-proxy": false
}
```

Don't forget to restart the docker service:
`sudo systemctl restart docker.service`


I don't know what the implications are of disabling the docker userland proxy.
It seems to simply use other methods of port forwarding, such as iptables. In that case it should be fine to do so.
